api/docs/architecture-decisions/0004-resource-name-prefixes.md

```diff
@@ -0,0 +1,41 @@
+# CF Resource GUID Prefixing
+
+Date: 2022-01-12
+
+## Status
+
+Draft
+
+## Context
+
+We have received feedback from the field that naming standard k8s typed objects (e.g. namespaces, statefulsets) with raw guids makes the system difficult for a k8s operator to use.
+
+For example:
+- it's hard when looking at a list of namespaces to tell which ones are CF-created. That makes it difficult to uninstall the things you installed to a cluster if you're just kicking the tires.
+- it's hard to tell which namespaces are cf spaces versus orgs. When listing other things like secrets or roles, it is hard to tell if they are in a space or an org or are totally unrelated to CF.
+- it's hard to be sure which stateful sets are CF apps, versus possibly other workloads that also have GUID names.
+
+The proposal to address this issue is to prefix the names of these resources with strings (e.g. "cforg-") that clearly identify them as CF resources. The request is to add this prefix to the object name rather than a label or other metadata, in order to make it easily discoverable.
```

Comment:
Minor:

I think having it be `cf-` (e.g. `cf-org-`, `cf-space-`, etc.), might be easier to parse visually (and perhaps programmatically for things like the HNC namespace allowlist).